crates/miden-testing/tests/agglayer/asset_conversion.rs

```diff
@@ -0,0 +1,246 @@
+        U256::from(1u64),
+    )
+    .await?;
+
+    // Test case 2: amount=1, scale to 1e18 (scale_exponent=18)
+    test_convert_to_u256_helper(
+        Felt::new(1),
+        Felt::new(18),
+        [0, 0, 0, 0, 0, 0, 232830643, 2808348672],
+        U256::from_dec_str("1000000000000000000").unwrap(),
+    )
+    .await?;
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_convert_to_u256_scaled_eth() -> anyhow::Result<()> {
+    // 100 units base 1e6
+    let miden_amount = Felt::new(100_000_000);
+
+    // scale to 1e18
+    let target_scale = Felt::new(12);
+
+    let asset_conversion_comp = asset_conversion_component(vec![]);
+    let asset_conversion_lib = asset_conversion_comp.library();
+
+    let script_code = format!(
+        "
+        use.std::sys
+        
+        begin
+            push.{}.{}
+            call.::convert_felt_to_u256_scaled
+            exec.sys::truncate_stack
+        end
+        ",
+        target_scale, miden_amount,
+    );
+
+    let program = Assembler::new(Arc::new(DefaultSourceManager::default()))
+        .with_debug_mode(true)
+        .with_dynamic_library(StdLibrary::default())
+        .unwrap()
+        .with_dynamic_library(asset_conversion_lib.clone())
+        .unwrap()
+        .assemble_program(&script_code)
+        .unwrap();
+
+    let exec_output = execute_program_with_default_host(program).await?;
+
+    let expected_result = U256::from_dec_str("100000000000000000000").unwrap();
+    let actual_result = stack_to_u256(&exec_output);
+
+    assert_eq!(actual_result, expected_result);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_convert_to_u256_scaled_large_amount() -> anyhow::Result<()> {
+    // 100,000,000 units (base 1e10)
+    let miden_amount = Felt::new(1000000000000000000);
+
+    // scale to base 1e18
+    let scale_exponent = Felt::new(8);
+
+    let asset_conversion_comp = asset_conversion_component(vec![]);
+    let asset_conversion_lib = asset_conversion_comp.library();
+
+    let script_code = format!(
+        "
+        use.std::sys
+        
+        begin
+            push.{}.{}
+
+            call.::convert_felt_to_u256_scaled
+            exec.sys::truncate_stack
+        end
+        ",
+        scale_exponent, miden_amount,
+    );
+
+    let program = Assembler::new(Arc::new(DefaultSourceManager::default()))
+        .with_debug_mode(true)
+        .with_dynamic_library(StdLibrary::default())
+        .unwrap()
+        .with_dynamic_library(asset_conversion_lib.clone())
+        .unwrap()
+        .assemble_program(&script_code)
+        .unwrap();
+
+    let exec_output = execute_program_with_default_host(program).await?;
+
+    let expected_result = U256::from_dec_str("100000000000000000000000000").unwrap();
+    let actual_result = stack_to_u256(&exec_output);
+
+    assert_eq!(actual_result, expected_result);
+
+    Ok(())
+}
+
+#[test]
+fn test_felts_to_u256_bytes_hand_encoded_values() {
+    // Test case 1: Simple sequential values 1,2,3,4,5,6,7,8
```

Comment:
I think the test name doesn't need to reflect the fact that these are hand picked